modules/Bio/EnsEMBL/Hive/Queen.pm

```diff
@@ -415,11 +416,14 @@ sub kill_all_workers {
                 $kill_status = 'requested successfully';
                 $worker->cause_of_death( 'KILLED_BY_USER' );
                 $self->register_worker_death( $worker );
+                if ( $worker->status ne 'SUBMITTED' ) {             # There is no worker_temp_directory before specialization
+                    $meadow->cleanup_temp_directory( $worker );
+                }
             }
         }
```

Comment:
Suggest adding
`else { $kill_status = 'could not kill, running under user ' . $worker->meadow_user }`
to provide a more complete message and avoid having `$kill_status` be undef

Reply:
Good point. I have fixed this in a new commit